packages/patternfly-4/react-table/src/components/Table/RowWrapper.js

```diff
@@ -1,23 +1,228 @@
 import React from 'react';
-import PropsType from 'prop-types';
+import { createPortal } from 'react-dom';
+import PropTypes from 'prop-types';
 import { tableExpandableRow, modifiers } from '@patternfly/patternfly-next/components/Table/table.css';
 import { css } from '@patternfly/react-styles';
+import { debounce } from '../../../internal/util';
+import TableConfirmButtonsRow from './TableConfirmButtonsRow';
+import { TableEditConfirmation } from './Table';
 
-const RowWrapper = ({ isOpen, isExpanded, ...props }) => {
-  return <tr {...props}
-    className={css(isExpanded !== undefined && tableExpandableRow, isExpanded && modifiers.expanded)}
-    hidden={isExpanded !== undefined && !isExpanded}
-  />;
+const hasConfirmationButtons = (editConfirmationType, { isEditing, isTableEditing, isFirst, isLast }) => {
+  switch (editConfirmationType) {
+    case TableEditConfirmation.TABLE_TOP:
+      return isTableEditing && isFirst;
+    case TableEditConfirmation.TABLE_BOTTOM:
+      return isTableEditing && isLast;
+    case TableEditConfirmation.ROW:
+      return isEditing;
+    case TableEditConfirmation.NONE:
+    default:
+      return false;
+  }
 };
 
-RowWrapper.propTypes = {
-  isOpen: PropsType.bool,
-  isExpanded: PropsType.bool
+export const areButtonsOnTop = (editConfirmationType, { isLast }) => {
+  switch (editConfirmationType) {
+    case TableEditConfirmation.TABLE_TOP:
+      return true;
+    case TableEditConfirmation.ROW:
+      return isLast;
+    case TableEditConfirmation.TABLE_BOTTOM:
+    case TableEditConfirmation.NONE:
+    default:
+      return false;
+  }
+};
+
+export const getEditStyles = (editConfirmationType, { isEditing, isTableEditing, isFirst, isLast }) => {
+  let editStyle;
+
+  if (isTableEditing) {
+    switch (editConfirmationType) {
+      case TableEditConfirmation.ROW:
+        if (isEditing) {
+          editStyle = {
+            background: '#def3ff',
```

Comment:
can you replace all color references with actual color tokens from `@patternfly/react-tokens` (ex consumption [here](https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-4/react-tokens#each-token-as-three-properties) )?

you can also find these and their default mappings here:
http://patternfly-react.surge.sh/patternfly-4/styles/tokens/

If these colors don't exist, maybe we can subtitute an existing [pf-next color](https://github.com/patternfly/patternfly-next/blob/5cf999d5e13eab30303729eabf8c2e26eaedadf7/src/patternfly/sass-utilities/colors.scss)?

Reply:
yes, I am planning to do the css  cleanup